test/functional/p2p_sendheaders.py

```diff
@@ -144,13 +144,6 @@ def send_getblocks(self, locator):
         getblocks_message.locator.vHave = locator
         self.send_message(getblocks_message)
 
-    def wait_for_getdata(self, hash_list, timeout=60):
-        if hash_list == []:
```

Comment:
This PR also removes this "do nothing if the `hash_list` is empty" behavior.  The `p2p_sendheaders` test still passes, but is there any reason we should keep the previous behavior?